src/pool/srv_pool.c

```diff
@@ -2627,7 +2630,10 @@ ds_pool_list_cont_handler(crt_rpc_t *rpc)
 	}
 
 out_free_cont_buf:
-	D_FREE(cont_buf);
+	if (cont_buf) {
+		D_FREE(cont_buf);
+		cont_buf = NULL;
```

Comment:
Hmm. This pointer will be nullified by D_FREE macro... Why we need the second one?

Reply:
I agree, I was just backing out the change from the PR as it's not directly related to the memory leaks.